### ISSUE TITLE
migrate : `record not found`

### ISSUE THREAD

**Issue by audibleblink** [BishopFox/sliver#1532]
**Describe the bug**
Unable to migrate with pid or process name

**To Reproduce**
1. Be on b4cf8482c1b2bfd79aba16f857df8537ba7ef830
1. Go to existing implant/session
2. `migrate -p 1084 ""` 
  a. ^ empty string required until #1531 is merged
4. See error from `sliver.log`
```
INFO[2023-12-29T15:57:06-05:00] [sliver/server/db/logger.go:20] github.com/bishopfox/sliver/server/db/helpers.go:196 record not found
[0.304ms] [rows:0] SELECT * FROM `implant_builds` WHERE `implant_builds`.`name` = "." ORDER BY `implant_builds`.`id` LIMIT 1
INFO[2023-12-29T15:57:06-05:00] [sliver/server/certs/certs.go:140] Generating TLS certificate (ECC) for 'HELPFUL_BOOKCASE' ...
INFO[2023-12-29T15:57:06-05:00] [sliver/server/certs/certs.go:65] Saving certificate for cn = 'HELPFUL_BOOKCASE'
ERRO[2023-12-29T15:57:06-05:00] [github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/logging/logrus/options.go:224] finished unary call with code Unknown
```
**Expected behavior**
Expect process to migrate

**Screenshots**
supporting convo
https://bloodhoundhq.slack.com/archives/CF84C5ZAQ/p1703883097368899

**Desktop (please complete the following information):**
 - OS: linux
 - Version debian 6.3.7

**Additional context**

**Comment by RafBishopFox** [comment]
@audibleblink - Thanks for reporting this. I think I found the issue, and the PR has been merged into `master`, so please give it a try.

**Comment by audibleblink** [comment]
@RafBishopFox the error when attempting to migrate while on 67f92d6b4

```
INFO[2024-01-03T14:23:18-05:00] [sliver/server/certs/certs.go:140] Generating TLS certificate (ECC) for 'EXCITING_QUINCE' ...
INFO[2024-01-03T14:23:18-05:00] [sliver/server/certs/certs.go:65] Saving certificate for cn = 'EXCITING_QUINCE'
INFO[2024-01-03T14:23:18-05:00] [sliver/server/c2/http.go:414] 10.10.110.254:27436 - /bundles/scripts/script/scripts/route.js?e=46110438g6204 -
ERRO[2024-01-03T14:23:18-05:00] [github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/logging/logrus/options.go:224] finished unary call with code Unknown
```

**Comment by RafBishopFox** [comment]
Hey @audibleblink - Could you help me replicate what you are working with? Migrating seems to work fine for me. Here is my setup:
```
sliver > sessions

 ID         Name                  Transport   Remote Address       Hostname          Username   Operating System   Locale   Last Message                               Health  
========== ===================== =========== ==================== ================= ========== ================== ======== ========================================== =========
 5cac77e5   IMAGINATIVE_STATION   mtls        10.11.12.132:57754   DESKTOP-R0AA4IE   User       windows/amd64      en-US    Thu Jan  4 11:03:33 UTC 2024 (1m51s ago)   [ALIVE] 
```

I opened `notepad.exe` on the machine with PID `4836`.

```
sliver > use 5cac77e5 

[*] Active session IMAGINATIVE_STATION (5cac77e5-7047-49b3-b7d6-9d9ae7f28b80)

sliver (IMAGINATIVE_STATION) > migrate -p 4836

[*] Successfully migrated to 4836


[*] Session 76a90bc8 IMAGINATIVE_STATION - 10.11.12.132:57766 (DESKTOP-R0AA4IE) - windows/amd64 - Thu, 04 Jan 2024 11:08:34 UTC
```

This is what it shows in the log:
```
INFO[2024-01-04T11:08:34-00:00] [github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/logging/logrus/options.go:220] finished unary call with code OK 
INFO[2024-01-04T11:08:34-00:00] [sliver/server/c2/mtls.go:88] Accepted incoming connection: 10.11.12.132:57766 
```

I see that you might be using HTTP(S) as the transport, so I generated an implant that uses HTTP, and I could migrate with that one just fine as well.

**Comment by audibleblink** [comment]
yeah you bet @RafBishopFox 

same behavior between both pid and name flags

i used `make linux-amd64` to generate the new bins after your changes were merged

i haven't tried deleting `~/.sliver`, wasn't sure if it was necessary, but if you need me to troubleshoot from a known-good state, i can do that

```
sliver > jobs

┌────┬───────────┬──────────┬───────┬─────────┐
│ ID │ NAME      │ PROTOCOL │ PORT  │ DOMAINS │
├────┼───────────┼──────────┼───────┼─────────┤
│ 1  │ https     │ tcp      │ 443   │         │
│ 2  │ grpc/mtls │ tcp      │ 31337 │         │
└────┴───────────┴──────────┴───────┴─────────┘

sliver > profiles

┌──────────────┬──────────────┬───────────────┬─────────────────────────────────────────┬───────┬────────────┬─────────────┬─────────────┬────────────┐
│ PROFILE NAME │ IMPLANT TYPE │ PLATFORM      │ COMMAND & CONTROL                       │ DEBUG │ FORMAT     │ OBFUSCATION │ LIMITATIONS │ C2 PROFILE │
├──────────────┼──────────────┼───────────────┼─────────────────────────────────────────┼───────┼────────────┼─────────────┼─────────────┼────────────┤
│ dll          │ session      │ windows/amd64 │ [1] https://10.10.14.22                 │ false │ SHARED_LIB │ enabled     │             │ default    │
│ generic      │ session      │ windows/amd64 │ [1] https://10.10.14.22                 │ false │ EXECUTABLE │ enabled     │             │ default    │
│ pivotsvc     │ session      │ windows/amd64 │ [1] namedpipe://10.10.121.107/pipe/alex │ false │ SERVICE    │ enabled     │             │ default    │
│ svc          │ session      │ windows/amd64 │ [1] https://10.10.14.22                 │ false │ SERVICE    │ enabled     │             │ default    │
└──────────────┴──────────────┴───────────────┴─────────────────────────────────────────┴───────┴────────────┴─────────────┴─────────────┴────────────┘

sliver > implants

┌────────────────────┬──────────────┬──────────┬───────────────┬────────────┬─────────────────────────────────────────┬───────┬───────────┬───────┬───────┐
│ NAME               │ IMPLANT TYPE │ TEMPLATE │ OS/ARCH       │     FORMAT │ COMMAND & CONTROL                       │ DEBUG │ C2 CONFIG │ ID    │ STAGE │
├────────────────────┼──────────────┼──────────┼───────────────┼────────────┼─────────────────────────────────────────┼───────┼───────────┼───────┼───────┤
│ EXCITING_QUINCE    │ session      │ sliver   │ windows/amd64 │    SERVICE │ [1] https://10.10.14.22                 │ false │ default   │ 12838 │ false │
│ INTERIM_THEISM     │ session      │ sliver   │ windows/amd64 │    SERVICE │ [1] https://10.10.14.22                 │ false │ default   │ 40174 │ false │
└────────────────────┴──────────────┴──────────┴───────────────┴────────────┴─────────────────────────────────────────┴───────┴───────────┴───────┴───────┘



[*] Session bf96472b EXCITING_QUINCE - 10.10.110.254:24553 (ws01) - windows/amd64 - Thu, 04 Jan 2024 14:50:01 EST

[*] Session 65605f97 INTERIM_THEISM - 10.10.110.254:7764 (ws03) - windows/amd64 - Thu, 04 Jan 2024 14:50:01 EST

sliver > use

? Select a session or beacon: SESSION  65605f97  INTERIM_THEISM   10.10.110.254:7764   ws03  NT AUTHORITY\SYSTEM  windows/amd64
[*] Active session INTERIM_THEISM (65605f97-7e69-4f84-8a8e-c7f9b65b7e4e)

sliver (INTERIM_THEISM) > migrate -n SearchApp.exe

[!] Error: rpc error: code = Unknown desc = record not found
```

**Comment by RafBishopFox** [comment]
@audibleblink - Ah okay! I was finally able to recreate what you are seeing. I had been testing with normal executables, not service executables. When I generated a service executable, I saw the same thing you are seeing. I think I figured out the issues with trying to migrate a Windows service implant and have submitted a PR to address it. Migration will take a moment because a new build has to be generated, but it should work.
